src/manager/identifiers.js

```diff
@@ -1,73 +1,113 @@
 import * as Ulid from '@kiosked/ulid'
 import * as emitter from '../utils/emitter'
-import { getCookie, getFromLs, getOrAddWithExpiration, setCookie } from '../utils/storage'
-import { isIframe, loadedDomain, parentHostname } from '../utils/page'
+import { loadedDomain } from '../utils/page'
 import { getLegacyId, getLegacyIdentifierKey } from '../utils/legacy'
 import { domainHash } from '../utils/hash'
+import { expiresInDays, strEqualsIgnoreCase } from '../utils/types'
+import { StorageStrategy } from '../model/storage-strategy'
 
-const NEXT_GEN_FP_NAME = '_lc2_duid'
+const NEXT_GEN_FP_NAME = '_lc2_fpi'
 const TLD_CACHE_KEY = '_li_dcdm_c'
-
 const DEFAULT_EXPIRATION_DAYS = 730
 
 /**
- *
- * @returns {string}
+ * @param {State} state
+ * @param {StorageHandler} storageHandler
  */
-export function determineTld () {
-  const cachedDomain = getCookie(TLD_CACHE_KEY)
-  if (cachedDomain) {
-    return cachedDomain
-  }
-  const domain = loadedDomain()
-  const arr = domain.split('.').reverse()
-  for (let i = 1; i < arr.length; i++) {
-    const newD = `.${arr.slice(0, i).reverse().join('.')}`
-    setCookie(TLD_CACHE_KEY, newD, { domain: newD })
-    if (getCookie(TLD_CACHE_KEY)) {
-      return newD
+export function resolve (state, storageHandler) {
+  try {
+    console.log('identifiers.resolve', state, storageHandler)
+
+    const determineTld = () => {
+      const cachedDomain = storageHandler.getCookie(TLD_CACHE_KEY)
+      if (cachedDomain) {
+        return cachedDomain
+      }
+      const domain = loadedDomain()
+      const arr = domain.split('.').reverse()
+      for (let i = 1; i < arr.length; i++) {
+        const newD = `.${arr.slice(0, i).reverse().join('.')}`
+        storageHandler.setCookie(TLD_CACHE_KEY, newD, undefined, undefined, newD)
+        if (storageHandler.getCookie(TLD_CACHE_KEY)) {
+          return newD
+        }
+      }
+      return `.${domain}`
     }
-  }
-  return `.${domain}`
-}
 
-/**
- * @return {LegacyId|null|undefined}
- * @private
- */
-function _legacyDuid () {
-  const _legacyEntry = getFromLs(getLegacyIdentifierKey())
-  return getLegacyId(_legacyEntry)
-}
+    const addDays = (days) => new Date().getTime() + (days * 864e5)
```

Comment:
did we try to change the variables with a function value to have only functions instead? Like
```
function addDays(day) {...}
```

It does not seem to be a problem to have nested functions inside the `resolve`.

Reply:
We don't want functions inside of functions, that's why it's like this.